crates/lint/src/sol/info/mixed_case.rs

```diff
@@ -54,11 +54,17 @@ impl<'ast> EarlyLintPass<'ast> for MixedCaseVariable {
 ///
 /// To avoid false positives like `fn increment()` or `uint256 counter`,
 /// lowercase strings are treated as mixedCase.
-pub fn is_mixed_case(s: &str) -> bool {
+pub fn is_mixed_case(s: &str, is_fn: bool) -> bool {
     if s.len() <= 1 {
         return true;
     }
 
     // Remove leading/trailing underscores like `heck` does
-    s.trim_matches('_') == format!("{}", heck::AsLowerCamelCase(s)).as_str()
+    if s.trim_matches('_') == format!("{}", heck::AsLowerCamelCase(s)).as_str() {
+        return true
+    }
+
+    // Ignore `fn test*`, `fn invariant*`, and `fn statefulFuzz` patterns, as they usually contain
+    // (allowed) underscores.
+    is_fn && (s.starts_with("test") || s.starts_with("invariant") || s.starts_with("statefulFuzz"))
```

Comment:
@grandizzy do you mean we should disable `mixedCase` for `invariant_*` but enforce it for `invariant*" (excluding `invariant_`)?

Reply:
yeah, that was my thought, wdyt?

Reply:
that makes sense to me yeah, so this logic needs to be updated